### PR TITLE
Add docs for all available utilities

### LINKED ISSUE #139: Add `import_string` to docs
### What was wrong?

The `import_string` function isn't documented.

### How can it be fixed?

Add basic docs.

### PR DESCRIPTION
**Pull request by reedsa**
Document utils for conversions, decorators, exception and module loading.
`text_if_str` `hexstr_if_str`, etc.

### What was wrong?

We were missing docs for several utility functions.

Closes #139

### How was it fixed?

Filled in docs for utilities that were undocumented.

### Todo:
- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-utils/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fselmo — APPROVED** [review]
Thanks for getting this in 👌🏼. I think we can improve on one of the examples to test for the ``AtributeError`` in the doctest itself, but nothing is really blocking here.

File: docs/utilities.rst [hunk 1]
```diff
@@ -928,6 +966,28 @@ As usual, instances create their own copy on assignment.
...
+Replaces `Old` exceptions in a method with `New` exceptions.
+Accepts a Dict, with `Old` exceptions pointing to 
+`New` exceptions.
+
+.. doctest::
+
+   >>> from eth_utils import replace_exceptions
+
+   >>> class Storage:
+   ...    val = 1
```
[the remark this answers is not in the document]
Comment (reedsa) [reply docs/utilities.rst:981]: Interesting find on the spaces. I learned that rst has a requirement that "footnotes" start with at least three spaces. If we are careful, 4 space indentation for everything sounds great. I'll go ahead and make those changes in another commit.

rst docs - https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#indentation
good example in the sphinx docs under this section - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#directives

[the remark this answers is not in the document]
Comment (fselmo) [reply docs/utilities.rst]: Come to think of it, we can make that example even simpler if we want.

```py
.. doctest::

    >>> from eth_utils import replace_exceptions

    >>> @replace_exceptions({TypeError: AttributeError})
    ... def thing(self):
    ...     if True:
    ...         raise TypeError

    >>> thing()
    Traceback (most recent call last):
       ...
    AttributeError
```

**Comment by fselmo** [comment]
@reedsa Looks like even though the docs build, there's a table formatting error after making the spacing more homogenous. 

When running `make build-docs`:

`.../eth-utils/docs/utilities.rst:811: ERROR: Malformed table.`

We could probably leave that table alone in that commit.

File: .build/__local__.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"contractTypes":{},"manifest":"ethpm/3","sources":{}}
```
Comment (fselmo) [inline .build/__local__.json:1]: These local settings should be added to the `.gitignore`, yeah?

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (fselmo) [inline .vscode/settings.json:1]: Same question here, was this intentional for some reason?

File: .build/__local__.json (hunk 2 above)
Comment (pacrob) [reply .build/__local__.json:1]: `build` is there, but not `.build`. Added to template for future propagation

File: .readthedocs.yml [hunk 4]
```diff
@@ -6,7 +6,8 @@ build:
     python: "3.11"
 
 sphinx:
-   configuration: docs/conf.py
+  configuration: docs/conf.py
+  fail_on_warning: true
```
Comment (pacrob) [inline .readthedocs.yml:10]: like!

**Pull request merged**